pkg/plugins/pfs/decl_parser.go

```diff
@@ -24,7 +23,9 @@ func NewDeclParser(rt *thema.Runtime, skip map[string]bool) *declParser {
 	}
 }
 
+// TODO convert this to be the new parser for Tree
```

Comment:
Something we can do after this PR is merged but I think we can refactor so this is moved to the kindsys pakage and we return something similar to PluginDecl from the kindsys instead.

Reply:
yeah `PluginDecl` should be able to go away completely now. There's a lot of simplification we can do with the types we have, now that the overall system is unified